File: jetpack/src/main/java/me/destro/android/playground/jetpack/databinding/MainActivity.java
package me.destro.android.playground.jetpack.databinding;

import android.content.Intent;
import androidx.databinding.DataBindingUtil;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import me.destro.android.playground.jetpack.R;
import me.destro.android.playground.jetpack.databinding.basic.BasicActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityExamplesBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_examples);


        binding.basic.getRoot().setOnClickListener(view -> {
            Intent intent = new Intent(this, BasicActivity.class);
            startActivity(intent);
        });

        binding.basic.title.setText(R.string.basic_example_title);
        binding.basic.description.setText(R.string.basic_example_description);
    }
}

File: jetpack/src/main/java/me/destro/android/playground/jetpack/navigation/MainActivity.kt
package me.destro.android.playground.jetpack.navigation

import android.content.Intent
import androidx.databinding.DataBindingUtil
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import me.destro.android.playground.jetpack.navigation.fullexample.FullExampleActivity
import me.destro.android.playground.jetpack.R
import me.destro.android.playground.jetpack.databinding.ActivityExamplesBinding
import me.destro.android.playground.jetpack.databinding.ActivityNavigationHostBinding

public class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityNavigationHostBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityExamplesBinding>(this, R.layout.activity_examples)

        binding.basic.root.setOnClickListener { view ->
            val intent = Intent(this, FullExampleActivity::class.java)
            startActivity(intent)
        }

        binding.basic.title.text = "Full example"
        binding.basic.description.text = "Description"
    }

}
